### PR TITLE
weather issue fixed and usertext issue fixed

### LINKED ISSUE #81: Two minor bugs found
- [ ] When opening a new tab with a theme that's not the default one, it'll show the default one for a few frames
- [x] When editing the text, it'll bring you to the start each second

https://github.com/user-attachments/assets/a1f4982f-bb8c-4920-aa4f-89e371597f7b

### PR DESCRIPTION
**Pull request by Minuga-RC**
#81 fixed (theme issue not fixed)
#84 fixed

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -336,15 +344,16 @@ function displayClock() {
         digitalClock.style.display = 'block';  // Show the digital clock
         analogClock.style.display = 'none';     // Hide the analog clock
     }
+   
 }
-setInterval((updated) => {
+function usertextupdate(){
```
Comment (ZiClaud) [inline script.js:349]: Could you change `usertextupdate()` to `updateUserText()` to make it more readable?
Everything else looks good

Comment (Minuga-RC) [reply script.js:349]: will do tmr

Comment (Minuga-RC) [reply script.js]: done

**Comment by Minuga-RC** [comment]
@XengShi ready to merge

**Review by ZiClaud — APPROVED** [review]

**Review by prem12321kumar — CHANGES_REQUESTED** [review]
line 79:

// Try to get the savedLocation or a previously stored location from localStorage
var currentUserLocation = savedLocation || localStorage.getItem("locationQ");

if (!currentUserLocation) {
    // If neither savedLocation nor localStorage has a location, fetch the IP-based location
    const geoLocation = 'https://ipinfo.io/json/';
    const locationData = await fetch(geoLocation);
    const parsedLocation = await locationData.json();
    currentUserLocation = parsedLocation.ip; // Update to the fetched IP-based location
    localStorage.setItem("locationQ", currentUserLocation); // Store the IP-based location in localStorage
}

var currentLanguage = getLanguageStatus('selectedLanguage') || 'en';

// Weather API call using currentUserLocation, which is either user input, localStorage, or IP address
const weatherApi = `https://api.weatherapi.com/v1/current.json?key=${apiKey}&q=${currentUserLocation}&aqi=no&lang=${currentLanguage}`;

const data = await fetch(weatherApi);
const parsedData = await data.json();

**Comment by prem12321kumar** [comment]
Conflict with my #91, @Minuga-RC this is my suggestion, it merges both of our ideas.

**Comment by Minuga-RC** [comment]
I don't really have time to do anything atm I'm - busy here making the rewrite

**Pull request closed without merging**